src/main/java/com/example/ajaxonboardingproject/config/NatsListenerBeanPostProcessor.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.ajaxonboardingproject.config
+
+import com.example.ajaxonboardingproject.controller.NatsController
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+
+@Component
+class NatsListenerBeanPostProcessor : BeanPostProcessor {
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any? {
+        if (bean is NatsController<*>) {
+            bean.connection.subscribe(bean.subject)
+            val dispatcher = bean.connection.createDispatcher { message ->
+                val method = bean.javaClass.methods.first { it.name == "generateReplyForNatsRequest" }
+                val response = method.invoke(bean, message) as ByteArray
```

Comment:
You process only beans with type `NatsController`. So, you can add to the interface function `generateReplyForNatsRequest`, then invoke it without reflection.

Reply:
let's rename `generateReplyForNatsRequest` -> `handleRequest`

Reply:
Done!